api/src/main/java/io/jsonwebtoken/SignatureAlgorithm.java

```diff
@@ -654,4 +661,14 @@ public static SignatureAlgorithm forName(String value) throws SignatureException
 
         throw new SignatureException("Unsupported signature algorithm '" + value + "'");
     }
+
+    private static Map<String, SignatureAlgorithm> preferredHmacAlgorithmMap() {
+        SignatureAlgorithm[] algs = {HS256, HS384, HS512};
+        Map<String, SignatureAlgorithm> lookupMap = new HashMap<>(6);
+        for (SignatureAlgorithm alg : algs) {
+            lookupMap.put(alg.jcaName.toUpperCase(Locale.ENGLISH), alg);
+            lookupMap.put(alg.pkcs12Name.toUpperCase(Locale.ENGLISH), alg);
```

Comment:
FWIW, the pkcs12 name is dotted numerals - no alphabetic characters, so changing case isn't strictly necessary. ;). But leaving it the way you have it is probably a little more obvious/self-documenting and could theoretically handle any other names.

Reply:
Out of curiosity - is this concept of acceptable HMAC names used anywhere else given the `SignatureAlgorithm` API change in the `jwe` branch?  I'd like to ensure it exists in one place only to ensure we don't duplicate logic that might accidentally become out of sync.